src/server/node_services/node_allocator.js

```diff
@@ -64,6 +66,25 @@ async function refresh_system_alloc(system) {
     for (const bucket of Object.values(system.buckets_by_name)) {
         tiering_list.push(bucket.tiering);
 
+        // realted to https://bugzilla.redhat.com/show_bug.cgi?id=1839117
+        // print information in case bucket.tiering is not iterable. should happen when the tiering is deleted and the bucket is not
+        if (!bucket.tiering.tiers) {
+            if (mongo_utils.is_object_id(bucket.tiering)) {
+                try {
+                    const deleted_tiering = await system_store.data.get_by_id_include_deleted(bucket.tiering, 'tieringpolicies');
+                    dbg.error(`bucket.tiering.tiers is undefined\\null. bucket=${
+                        util.inspect(bucket, { depth: 5 })
+                    } tiering_policy=${
+                        util.inspect(deleted_tiering, { depth: 5 })
+                    }`);
+                } catch (err) {
+                    dbg.error(err);
+                }
+            } else {
+                dbg.error(`bucket.tiering.tiers is undefined\\null. bucket=`, bucket);
```

Comment:
```suggestion
                dbg.error(`bucket.tiering.tiers is undefined\\null. bucket=`, bucket, bucket.tiering);
```

Reply:
added util.inspect. tiering should be printed as part of the bucket